### ISSUE TITLE
Editorial onward journey component: horizontally scrollable promos

### ISSUE THREAD

**Issue by jroebu14** [bbc/simorgh#9697]
**Is your feature request related to a problem? Please describe.**
As part of our objective to scale Optimo publishing to editorial users, we want to support the editorial teams in their ability to link to other stories, videos or content that are relevant to the narrative or provide deeper context, understanding and analysis around a story.

We would like to build an editorial onward journey component for a set of links for Optimo articles.

**Describe the solution you'd like**
A new component that consumes Optimo data and renders a maximum of 3 link promos with event tracking.

<details>
<summary>Background</summary>

We want to enable editorial users to share links within the body of an article without completely disrupting the user flow.

There will be an editorially curated “Links” block within Optimo that can be blended with other onward journeys in the future. 

The "Links" block only contains a set of 'Link' types containing text and images only

This block will serve a similar purpose to both the see-alsos and the bulleted hyperlinks in stories.

Editorial can place the block or blocks anywhere within the article. There are no limits on now many can be used.

The blocks will be considered as part of the article and the  components are rendered as part of the article

The links block can have titles set in Optimo

There will be 2 components powered by this block on the page : - 'more on this story' and 'Related content'.

Test Optimo (Optimo work in progress. This is shown as example)

Link to article with `links` blocks: https://www.test.bbc.com/pidgin/articles/crrrkxz2k0ko.json

</details>

<details>
<summary>Design</summary>

<img width="497" alt="Screenshot 2022-01-12 at 11 57 42" src="https://user-images.githubusercontent.com/90621252/149137731-3bb5e95f-8979-4213-b0fc-ec6983408f66.png">

Refer to NEWSWORLDSERVICE-1223 for more info on designs

</details>

**Developer notes**
- Develop with RTL/LTR languages in mind and ensure horizontal scroll scrolls in correct direction for languages
- Render null if no promos are contained in the block

**Questions**
- Should we split event tracking into separate issue? (DR: Yes)
- Do we use CSS line-clamp for truncating to 3 lines? This isn't supported in older browsers but this may be ok to let these lines flow to +3 lines in older browsers (DR: It is ok for older browser to flow 3+ lines)
- Ask UX about the width of promo when links are horizontally scrollable e.g. set amount of pixels or percentage of viewport? (DR: set amount of pixels at different viewports)
- Do we have a minimum promo box size for very small devices? (DR: No for 2 reasons: Might be an edge case we don't care about and because this is mitigated by the scrollable nature of the component)

**a11y notes**
- Please refer to NEWSWORLDSERVICE-1223 for up to date a11y acceptance criteria

**Testing notes**
Test page available 
https://www.test.bbc.com/pidgin/articles/crrrkxz2k0ko
https://www.test.bbc.com/pidgin/articles/crrrkxz2k0ko.json

RtoL
https://www.test.bbc.com/arabic/articles/cjn2e0jek55o
https://www.test.bbc.com/arabic/articles/cjn2e0jek55o.json

- Test mobile and desktop
- Test RTL/LTR
- Test truncation
- Test number of promos to test the 3 maximum e.g. test with 2 promos, test with +3 promos
- Test tabbing order
- Test viewport size and promo comes into view when tabbed to

Dev insight: Will Cypress tests be required or are unit tests sufficient? Will there be any potential regression? etc

- [x] This feature is expected to need manual testing.

**Checklist**
- [x] (BBC contributors only) This issue follows the [repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)

**Additional context**

**Comment by skwong25** [comment]
Testing carried out on all Level A & B browsers: 

LEVEL A: 
* **Chrome on mobile and desktop**
    * Latest (pass)
    * Desktop (pass)
    * Mobile (pass) 
* **Safari on Last 2 IOS versions** (pass)
    * Safari 13.x (pass)
    * Safari 14.x   (pass)
* **Opera Mini (Android)** (pass) 
* **Latest Opera on Desktop** (pass)
* **Latest Firefox on Desktop** (pass) 
* **Latest Samsung Browser** (Android) (pass)

LEVEL B:
* **Chrome 38.x+ on mobile and desktop** (Agreed with Product that 38.x to be excluded until browser list updated.) 
    * Desktop tested on 78.x (pass) 
    * Mobile tested on 75.x. (pass)

* **UC Browser Mini** (latest) - Only UC Browser (Android, latest) is available (pass).

-----------------------------------------------

Issues noticed on Level A: 
*  **Opera Mini Extreme Mode**
    * Pidgin - horizontal scroll does not work 
    * Arabic (RTL script) - horizontal scroll does not work 

Issues noticed on Level B: 
* **Firefox 48 on Desktop**
     * Pidgin - content is right-aligned and third card is hidden, horizontal scroll bar is present but not obvious 
     * Arabic (RTL script) - content looks left-aligned and third card is hidden, horizontal scroll bar is present but not obvious. Card text overflowing incorrectly (see screenshot 1 below)
* **Edge 18.x on Desktop** (Windows 10)
    * Card text does not wrap but overflows without the ‘…’ that usually denotes ellipsis. (see screenshot 2 below)

<img width="1791" alt="Desktop Firefox 48 RTL " src="https://user-images.githubusercontent.com/59804681/149376604-03fb5ca7-de45-46e9-a3d1-c56415c5704d.png">

<img width="1787" alt="Desktop Windows 10 Edge 18 x - ellipsis " src="https://user-images.githubusercontent.com/59804681/149376885-3ffe730b-32a5-4a62-9a55-2004dfb8be63.png">
